### PR TITLE
Crawl cloudformation stacks and resources

### PR DESCRIPTION
**Pull request by sihil**
## What does this change?
There is a rich seam of information in the cloudformation stacks. This extracts it every few minutes and makes it available to prism clients. With this we can more easily ask questions about resources that are or are not looked after by cloudformation and identify the stack that a resource belonds to more quickly.

## How to test
This should be tested in CODE to check that is works as expected and doesn't reveal private data.

## How can we measure success?
 - Being able to identify all S3 buckets not created in Cloudformation.
 - Being able to identify any CFN stacks that have stack policies

## Have we considered potential risks?
There are a few risks worth a little more thinking about:
 - will the extra data make prism too slow again?
 - is any of the data in outputs and parameters for stacks be private?

### CODE REVIEW

File: app/utils/filter.scala [hunk 1]
```diff
@@ -71,5 +77,8 @@ object ResourceFilter {
     ResourceFilter((defaultKeys ++ filterKeys).toMap)
   }
 
-  lazy val all: Matchable[JsValue] = (_: JsValue) => true
+  lazy val all: Matchable[JsValue] = new Matchable[JsValue] {
+    override def isMatch(value: JsValue): Boolean = true
+    override def inverse: Boolean = false
+  }
```
Comment (akash1810) [inline app/utils/filter.scala:83]: What are these changes for?

**Review by akash1810 — COMMENTED** [review]
The change set is a lot smaller than I imagined - its way too simple to crawl a new AWS resource!

I don't fully understand the changes to `filter.scala`

File: app/utils/filter.scala (hunk 1 above)
Comment (sihil) [reply app/utils/filter.scala:83]: ah - I knew there was something I still needed to test. I'll document it and ping you.

Comment (sihil) [reply app/utils/filter.scala]: I've now extracted the filtering changes into their own PR #214 for clarity and ability to independently roll either back.

**Review by akash1810 — APPROVED** [review]
LGTM!

File: app/controllers/Api.scala [hunk 2]
```diff
@@ -223,6 +223,14 @@ class Api (cc: ControllerComponents, prismDataStore: Prism, prismConfiguration:
       Api.singleItem(prismDataStore.vpcAgent, arn)
     }
 
+    def cloudformationStackList = Action.async { implicit request =>
+      Api.itemList(prismDataStore.cloudformationStackAgent, "cloudformation-stacks")
```
Comment (akash1810) [inline app/controllers/Api.scala:227]: Spotting whilst testing on CODE - kebab casing here makes the `jq` query a bit annoying as you have to [wrap the field in quotes](https://stackoverflow.com/a/37344498/3868241):

`curl https://prism.../cloudformationStacks | jq '.data."cloudformation-stacks"[].outputs'`

Could this be `cloudformationStacks` instead?

`curl https://prism.../cloudformationStacks | jq '.data.cloudformationStacks[].outputs'`

Not a blocker.

**Comment by akash1810** [comment]
Shipping this as @sihil is on paternity leave.

- [ ] Update StackSet to add the additional permissions
- [ ] Deploy to CODE
- [ ] Check that no stack outputs secrets
- [ ] Deploy to PROD

**Review by akash1810 — APPROVED** [review]

**Comment by akash1810** [comment]
> is any of the data in outputs and parameters for stacks be private?

It's worth noting this PR explicitly sets these values to `None` in the response so we can ship this now whilst we get teams to update their stacks.

**Comment by akash1810** [comment]
I've deployed this branch to PROD w/out issues. Merging.

**Pull request merged**